### PR TITLE
Edits for query spec

### LINKED ISSUE #31: Address Query specification review comments
From Paul Knight:

- [ ] We have transitioned the OASIS web site and document repository to use HTTPS instead of HTTP for almost all URIs going forward. This actually introduces some transitory issues in references to older specifications. If older HTML specifications which use .css files (unfortunately, this includes OSLC specs) are referenced using HTTPS, then the .css scripts will not be executed, and the visible document formatting will be affected. What this means in practical terms is that the "Previous version" hyperlinks on the front page should continue to use HTTP the next time a new stage of an existing specification is published. Later, when the Previous version links point to a document published as using HTTPS, they can use HTTPS. We should be able to track and correct these issues for you.

jra: I don't think this will impact Open Project URIs and their use of css, but we may have one round of revisions where the Previous Version needs to be http.

- [x] You are using a somewhat obsolete structure for the IPR Policy statement, which occurs twice in each specification: first as the third paragraph in the "Status:" section on the front page, and second as the IPR Policy section (usually Section 1.1). Specific changes desired:  remove "being" from "This specification is being provided" in third paragraph of Status section, and then copy the entire paragraph to use as the IPR Policy Section (1.1).  So the text at both places should be:

This specification is provided under the RF on Limited Terms Mode of the OASIS IPR Policy, the mode chosen when the Technical Committee was established. For information on whether any patents have been disclosed that may be essential to implementing this specification, and any offers of patent licensing terms, please refer to the Intellectual Property Rights section of the TC’s web page (https://www.oasis-open.org/committees/oslc-core/ipr.php). 

jra: this no longer applies as the IPR policy is removed from open project specifications and replaced with a reference to the chosen license.

- [x] 3. Please ensure that the approved ZIP file contains all of the relevant artifacts.  For OSLC-Query, I had to go fishing in the TC's SVN repository to get the diagram at the top of Section 3. (images/CoreDiscoveryUML.png)

jra: the open project publishing process is as yet unknown. But it will likely still involve ReSpec rendering of static HTML and references to images and styles that will need to be included in the publish process. Hopefully we will be able to script the publish process to remove manual activities that can result in errors.

- [x] 4. Widely applicable across the OSLC specifications - the web site at open-services.net has apparently reorganized the storage of all of the Version 2 OSLC files, so that all of the links to those documents are broken.  The fix to the links is fairly simple, since the documents appear to all be available at "https://archive.open-services.net/" instead of the address "http://open-services.net/".  I made this change to the references in OSLC-Query-v3.0. You will need to apply this to future publications of all OSLC documents which contain references to these V2.0 documents. If you have access to the open-services.net systems, you may be able to arrange for some kind of redirection at the site, to avoid reworking all of the OASIS documents.

jra: this should be fixed with the effort Andrew has just done to update all the redirects to the older specs in the archive.
  
- [x] 5. We have changed the OASIS headings text colors and logo (as in OSLC-Domains, for the OSLC-AM v2.1 specification [1]).  We'll work with you to get these changes into your future publications. It mostly involves including a new version of the "spec.css" file, and updating the reference to the logo file.

jra: this no longer applies as the open project has its own icon which is handled by ReSpec.

- [x] 6. "Nice-to-have" - If you are able to set up your ReSpec environment to generate a horizontal rule (line) using the  <hr> tag in the HTML, just before each major section, that will be helpful for publication going forward. (It is already being generated - at least in OSLC-Query - after the OASIS logo, before the "Notices" section, and before and after the Table of Contents. I had to add it before Section 2, 3, etc. In addition to following the OASIS style for HTML, this is also our trigger for a new page in the PDF, when we generate it from the HTML.

jra: this looks to be done as part of the open project ReSpec updates.

- [x] 7. I noted that there's a case where the same text appears in two different conformance clauses, which (in Section 10 of OSLC Query) causes the link for query-41 to jump to its duplicate, query-27 (which I just noted, does not show up in the table in Section 10). Hmmm....

I think that covers the items I noted.  I'm not submitting this as a formal Public Review comment, so there's no need for any specific response or comment-tracking. (Although maybe item 7 should be noted for some attention...) 

- [ ] Accessibility Checker: https://achecker.ca/checker/index.php when given the HTML version of OSLC Query Version 3.0 https://docs.oasis-open.org/oslc-core/oslc-query/v3.0/csprd01/oslc-query-v3.0-csprd01.html 

From Patrick Durusau <patrick@durusau.net>

The [TAB Issues](https://issues.oasis-open.org/projects/TAB/issues/TAB-1660?filter=allopenissues) are the formal review issues that must be addressed. See issues where affectsVersion = "OSLC Query Version 3.0". Use "OSLC Query Version 3.0" filter

It isn't necessary to acknowledge each comment separately. A single  email acknowledging this email will be sufficient.   If the TC chooses to ?defer? resolution of an issue to a later version,  please choose Defer Issue under Available Workflow Actions (after opening the issue).   When the TC has acted on these issues, I would appreciate a pointer to the TCs resolution as the TAB is tracking the resolution of comments from TAB members.

TAB issues that require action:

- [x] TAB-1658 Caption on all tables
- [x] TAB-1656 RDF/XML bib ref for XMLLiteral standard
- [x] TAB-1655 reference XML Schema in oslc.where clause
- [x] TAB-1654 incorrect citations, use  http://docs.oasis-open.org/templates/ietf-rfc-list/ietf-rfc-list.html http://docs.oasis-open.org/templates/w3c-recommendations-list/w3c-recommendations-list.html 
- [x] TAB-1653 non-normative terminology
- [x] TAB-1652 conformance clauses vs normative statements
- [x] TAB-1651 implementation vs server
- [x] TAB-1650 implementations of this specification = OSLC query capability implementations or OSLC Query server
- [x] TAB-1649 Section 3 reference to discovery image
- [x] TAB-1648 duplicate anchors in oslc.where clause table
- [ ] TAB-1647 fix permanent redirects

### LINKED ISSUE #132: Behaviour if oslc.orderBy specifies an unsupported ordering property
Some implementations might not support sorting by all queryable properties. If a client specifies a property that is not supported for sorting, the spec should define the behaviour.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-149 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #133: Query result container issues in Query spec
The description of the query result could do with at least one example.

The current specification does not describe any container, but related guidance documents do. The current position is that an oslc:QueryCapability may reference Zero-or-one oslc:resourceShape. If specified, that resource shape is the resource shape of the query result container, and it MUST denote at least one OSLC property as being a member property using the oslc:isMemberProperty. That predicate is defined in OSLC Core 3.0 Resource shapes, but was not described in the query spec. If a query capability does not reference a resource shape, then the current \[undocumented\] behaviour is to use rdfs:member.

Some implementations also provide a resource shape for the query result members by providing an oslc:valueShape for the OSLC property that has oslc:isMemberProperty set to true. However, the semantics of that referenced shape is ambiguous - see [https://issues.oasis-open.org/browse/OSLCCORE-145](https://issues.oasis-open.org/browse/OSLCCORE-145).

In many cases, both OSLC Core and OSLC Configuration Management provide other containers, and these are specified as LDPCs. These LDPCs have some similarity with a query base except that they do not support oslc.where or oslc.searcdhTerms. We should align the query result container with LDPCs. This has the advantage that it‘s backwards compatible with OSLC 2.0 query, but has the usual semantics of LDPCs widely used in OSLC Core 3.0.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-102 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #146: Query spec describes rdf:resource in example
[https://open-services.net/bin/view/Main/OSLCCoreSpecQuery](https://open-services.net/bin/view/Main/OSLCCoreSpecQuery) has the following text for an example:

"Property values will often refer to or be other resources (in RDF/XML the rdf:resource attribute is used to identify the resource). Suppose bugs are linked to testcases using a property name qm:testcase where the prefix qm: stands for the URI [http://qm.example.com/ns](http://qm.example.com/ns). The following query finds all bugs that are linked to testcase [http://example.com/tests/31459:](http://example.com/tests/31459:) "

I don‘t think it‘s appropriate for the spec to define an arcane referencce to rdf:resource that is solely a feature of RDF/XML serialization.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-109 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #149: What properties can be defined in the oslc.where of a query
In section "7.2 oslc where" (  
[https://tools.oasis-open.org/version-control/browse/wsvn/oslc-core/trunk/specs/oslc-query.html#oslc.where](https://tools.oasis-open.org/version-control/browse/wsvn/oslc-core/trunk/specs/oslc-query.html#oslc.where)), it is not clear what defines the set of properties that can be given in the oslc.where pattern.

Let‘s take the provided Example2 as a example, where cm:memberBug defines the membership property, and dcterms:identifier is used in the oslc.where query.

The example states "supposed bug reports have a dcterms:identifier" property....

But how do we know we are dealing with "bug reports"? I see 4 alternatives:

1\. The oslc:resourceType property of the Query Capability refers to Bug.

2\. The Query Capability "provides Resource Shapes that describe the property values that may be expected in the resources that are queryable via the query capability." (as stated under [https://open-services.net/bin/view/Main/OslcCoreSpecification#Query\_Capabilities](https://open-services.net/bin/view/Main/OslcCoreSpecification#Query\_Capabilities))  
This shape would have to be different from oslc:resourceShape property, which is to describe the baseQuery itself.  
(See [https://issues.oasis-open.org/browse/OSLCCORE-102](https://issues.oasis-open.org/browse/OSLCCORE-102) for discussion on query shapes)

3\. The membership property cm:memberBug has a Range defined, that refers to Bug.

4\. oslc.where can refer to any properties that don‘t even need to be part of the Bug resource.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-126 (opened by @jadelkhoury; previously assigned to @jamsden)_

### LINKED ISSUE #150: OSLC Query BNF for wildcards seems questionable
I was about to answer this question [https://jazz.net/forum/questions/227587/how-to-use-rm-apioslc-where-clause-using-not-equal-null-conditions](https://jazz.net/forum/questions/227587/how-to-use-rm-apioslc-where-clause-using-not-equal-null-conditions) with the link to the OSLC Query spec but then I saw !="\*" on the linked docs in [https://www.ibm.com/support/knowledgecenter/SSZRHJ/com.ibm.mif.doc/gp\_intfrmwk/oslc/r\_oslc\_query\_params.html](https://www.ibm.com/support/knowledgecenter/SSZRHJ/com.ibm.mif.doc/gp\_intfrmwk/oslc/r\_oslc\_query\_params.html).

That made me read BNF extra carefully and from the shortened subset below you can see my concern:

oslc\_where ::= "oslc.where=" compound\_term  
compound\_term ::= simple\_term (space? boolean\_op space? simple\_term)\*  
simple\_term ::= term | scoped\_term  
space ::= " " /\* a space character \*/  
boolean\_op ::= "and"  
term ::= identifier\_wc comparison\_op value | identifier\_wc space in\_op space? in\_val  
scoped\_term ::= identifier\_wc "

{" compound\_term "}

"  
identifier\_wc ::= identifier | wildcard

This query is valid:

oslc.where=\* > 10

While this one is invalid:

oslc.where=name=\*

Now that I look at my examples, this actually makes sense (leaving aside a question why would you filter by any property being greater than a certain number), while IBM manual‘s "The parent!="\*" query is semantically equivalent to the parent="NULL" query." does not!

Do you have any automatic BNF validators to check the sanity of the BNF further?

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-138 (opened by @berezovskyi; previously assigned to @jamsden)_

### LINKED ISSUE #151: OSLC Query doesn&apos;t describe paging
The OSLC Query spec does not define paging. The earlier and now obsolete OSLC simple query defined paging support using oslc.limit and oslc.offset.

OSLC Core 3.0 describes resource paging:  
1) LDPC paging is a SHOULD  
2) OSLC paging is a MAY.

However LDPC paging was defined after the OSLC Core 2.0 specs were published. If a server implements paging, it is most likely for query results. Query results are more likely to use OSLC paging in existing implementations. Does Core 3.0 need to be changed to reflect a changed order, or perhaps both #1 and #2 should be a MAY?

I think the query spec should discuss paging, but reference OSLC Core for the details. OSLC Core should reference the current LDPC paging information at [https://www.w3.org/2012/ldp/hg/ldp-paging.html](https://www.w3.org/2012/ldp/hg/ldp-paging.html). Since query results may contain additional triples from associated resources of member resources through the use of nested oslc.select properties, the spec should give some guidance as to how the page boundaries might be implemented.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-127 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #153: OSLC query does not specify how oslc.orderBy affects the results
OSLC Query 2.0 describes an oslc.orderBy query parameter. However, our returned results use unordered representations such as a list of statements with rdfs:member. If the query response is non-paged, then oslc.orderBy serves no purpose - a consumer cannot determine the order from the returned RDF.

It potentially serves a purpose for a paged response (either LDPC paging, or OSLC paging), where the first page should contain a subset of the sorted members starting at the beginning and so on. The OSLC Query 2.0 does not describe this usage pattern so it‘s unclear whether anything more was intended.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-143 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #158: OSLC Core 3.0 Discovery does not describe the results of GET on a QueryCapability queryBase URL.
OSLC Core 3.0 Discovery defines QueryCapability and oslc:queryBase: The base URI to use for queries. Queries are invoked via HTTP GET on a query URI formed by appending a key=value pair to the base URI, as described in Query Capabilities section. But there is no Query Capabilities section in the Discovery document that describes what the results of such a GET would be, or addresses the OSLC Core 2.0 use of rdfs:member vs. ldp:contains.

[https://web.archive.org/web/20151031160403/http://open-services.net/bin/view/Main/OSLCCoreSpecRDFXMLExamples#Specifying\_the\_shape\_of\_a\_query](https://web.archive.org/web/20151031160403/http://open-services.net/bin/view/Main/OSLCCoreSpecRDFXMLExamples#Specifying\_the\_shape\_of\_a\_query) provides examples that use rdfs:member. However, that documernt does not appear to be referenced by the specification.

The use of lpd:contains is preferable because it is consistent with other LDPCs, some of which are, in essence, query-based containers without support for oslc.where. However, rdfs:member appears to be the current query 2.0 usage, albeit not described in the OSLC Query 2.0 specification itself.

[https://web.archive.org/web/20151031160403/http://open-services.net/bin/view/Main/OSLCCoreSpecRDFXMLExamples#Specifying\_the\_shape\_of\_a\_query](https://web.archive.org/web/20151031160403/http://open-services.net/bin/view/Main/OSLCCoreSpecRDFXMLExamples#Specifying\_the\_shape\_of\_a\_query) also states:

* * *

Specify both via Query Resource Shape. In the Service resource in the definition of the Query Capability, add an oslc:resourceShape property-value to specify a complete Resource Shape that defines the shape of the query. The shape should specify both a resource type to be used for the results, and a member property to be used to represent individual query results. This will enable people and query builders to discover query-able fields and the shape specifies the form that will be returned.

* * *

However, although a resource shape can have more than one oslc:describes, that results in the referenced oslc properties applying to all those types. One way to avoiid that is to have a queryCapability reference two resource shjapes, one for the returned members, the other for the container. However, OSLC Core 2.0 specified oslc:Zero-or-one for the oslc:resourceShape of a query capability.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-140 (opened by @jamsden; previously assigned to @jamsden)_

### LINKED ISSUE #160: Semantics of query for equals a string value
The OSLC Query 2.0 specification does not explicitly describe whether an "=" operator on a string literal should be case-sensitive or case-insensitive. In most cases, a case-insenstive query is more accommodating to finding resources, especially since the spec does not support string pattern matching - see [https://issues.oasis-open.org/browse/OSLCCORE-157](https://issues.oasis-open.org/browse/OSLCCORE-157).

However, the spec references [https://www.w3.org/TR/rdf-sparql-query](https://www.w3.org/TR/rdf-sparql-query) and in that, string equals is case sensitive. This is perhaps one case where strict adherence to SPARQL for our abstract OSLC query makes it less usable to users.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-158 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #164: Support for string pattern matching
OSLC Query 2.0 does not define any means for a client to query a specified textual property for a partial match, such as "contains", or other string pattern-matching capabilities. I found a number of discussions in Jazz.net forums asking for such a feature. A common requirement is to query for change reqeusts whose summary/title contains a keyword. While oslc.searchTerms provides some means of finding change requests of interest, the result could contain results where text of interest is used in any textaul properties.

RTC appears to allow "\*" in a string literal value as a match anything.  
GCM supports the use of "%" and "\_" in string values and the "=" operator is then implemented using a case-insenstive SQL LIKE.

Since we have a few implementation attempting to address this requirement, I‘d like to see such a feature standardized but a MAY rather than a MUST.

The use of "%" and "\_" from SQL like is preferable because it is easily implementable on RDB based applications, and can easily be transformed into a SPARQL regular expression for RDF/SPARQL based applications.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-157 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #165: Behaviour if neither oslc.where nor oslc.searchTerms are specified
Currently, the OSLC Query 3.0 draft specification does not define the behaviour of OSLC Query when neither oslc.where nor oslc.searchTerms is specified. We should provide more guidance on this.

Some implementations might want to return all artifacts of the RDF type associated with the query capability. However, when the number of artifacts is very large, and especially when a server does not support paging, such a request could overwhelm a server, or the resultant large response might overwhelm the requesting client.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-179 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #174: Query spec does not define behaviour of oslc.where or oslc.searchTerms if not implemented
[https://open-services.net/bin/view/Main/OSLCCoreSpecQuery](https://open-services.net/bin/view/Main/OSLCCoreSpecQuery) describes two differenty types of query:  
1) A text search query specified by oslc.searchTerms  
2) A property graph based query specified by oslc.where

In order to ease the implementation buden of the spec, we should allow servers to implement one but not necessarily both. Currently a client has no means of discovering whether either or both are supported before issuing a query request.

The spec should define the behaviour if oslc.searchTerms or oslc.where is specified but the server does not support that capability.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-105 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #175: literal_value of the oslc_where syntax is not well-defined
The spec is not clear on how to interpret the literals w/o the xsd data type.

E.g.

The terms boolean and decimal are short forms for typed literals. For example, true is a short form for "true"^xsd:booleancode>, 42 is a short form for "42"xsd:integer and 3.14159 is a short form for "3.14159"^xsd:decimal.

does not specify how I am supposed to know whether 42 is an integer but 3.14 is a decimal (or a single-precision float?), let alone how I am supposed to ensure that ‘true‘ is a boolean True, not a "true" string literal.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-134 (opened by @berezovskyi; previously assigned to @DavidJHoney)_

### LINKED ISSUE #176: oslc.where syntax differs from Lyo reference implementation
[https://open-services.net/bin/view/Main/OSLCCoreSpecQuery](https://open-services.net/bin/view/Main/OSLCCoreSpecQuery) description of the syntax of oslc.where only supports:

value ::= uri\_ref\_esc | literal\_value  
uri\_ref\_esc ::= /\* an angle bracket-delimited URI reference in which > and \\ are -escaped. \*/

However, when I plaigerised the Antlr definition of this from Lyo, I found it also accepted prefixed names. For example, the following was supported:

oslc.where=dcterms:creator=jtsuser:John

providing that the prefix jtsuser was defined. This seems useful, so should we adopt that in the spec?

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-110 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #181: oslc:postBody does not indicate it should be used only for OSLC paging
oslc:postBody, in its shape and vocabulary entries, says that this property defines the body of a POST request to return the next page if the response was to a POST request, going on:

"Where a paged resource supports POST with an application/x-www-form-urlencoded body as an alternative to GET to avoid the request URI exceeding server limitations, the oslc:ResponseInfo in the response to the POST SHOULD contain this property so that a client knows what to POST to get the next page."

This may be applicable to OSLC 2.0 paging, but it is incompatible with LDP paging. LDP paging uses 303 redirects, which should be followed using a GET request. The http spec states:  
The 303 (See Other) status code indicates that the server is  
redirecting the user agent to a different resource, as indicated by a  
URI in the Location header field, which is intended to provide an  
indirect response to the original request. A user agent can perform  
a retrieval request targeting that URI (a GET or HEAD request if  
using HTTP), which might also be redirected, and present the eventual  
result as an answer to the original request. Note that the new URI  
in the Location header field is not considered equivalent to the  
effective request URI.

This status code is applicable to any HTTP method. It is primarily  
used to allow the output of a POST action to redirect the user agent  
to a selected resource, since doing so provides the information  
corresponding to the POST response in a form that can be separately  
identified, bookmarked, and cached, independent of the original  
request.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-184 (opened by @ndjc; previously assigned to @jamsden)_

### LINKED ISSUE #188: Query spec does not describe oslc.limit or oslc.offset, but references obsolete spec that does
The query spec does not mention oslc.limit or oslc.offset, but the obsolete document it references at [https://open-services.net/bin/view/Main/OslcSimpleQuerySemanticsV1](https://open-services.net/bin/view/Main/OslcSimpleQuerySemanticsV1) does. So it‘s not clear whether oslc.limit and/or oslc.offset are part of the spec or not. If not, the spec does not mention the possibility of a paged response.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-125 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #194: &#167;1.1 Terminology missing
§1.1 Terminology defines just a few placeholders.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-129 (opened by @berezovskyi; previously assigned to @DavidJHoney)_

### LINKED ISSUE #198: Difference between oslc.select and oslc.properties could be clearer
The spec describes the difference between oslc.select and oslc.properties. However, it‘s not as clear as it could be and would benefit from an example.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-122 (opened by @DavidJHoney; previously assigned to @jamsden)_

### LINKED ISSUE #205: Migrate OSLC 2.0 Query Specification to OASIS
This document needs to be migrated to OASIS from open-service.net, just like all the other OSLC 2.0 specifications. This should be added as another part of the OSLC Core 3.0 multi-part specification, and all references to OSLC query should be updated to refer to this migrated part.

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-89 (opened by @jamsden; previously assigned to @jamsden)_

### LINKED ISSUE #207: &#167;1.2 References may be moved to the bottom
Do we need it to be in the intro?

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-130 (opened by @berezovskyi; previously assigned to @DavidJHoney)_

### LINKED ISSUE #209: Namespace definition in &#167;5
Why define oslc\_core here?

---
_Migrated from https://issues.oasis-open.org/browse/OSLCCORE-131 (opened by @berezovskyi; previously assigned to @DavidJHoney)_

### LINKED ISSUE #285: Link to the Github repo from the spec drafts
From https://github.com/oasis-open-projects/administration/issues/8

### PR DESCRIPTION
**Pull request by DavidJHoney**
Editorial changes

Closes #198, Closes #209, Closes #285, Closes #209, Closes #207, Closes #205, Closes #198, Closes #194, Closes #188, Closes #181, Closes #176, Closes #175, Closes #174, Closes #165, Closes #164, Closes #160, Closes #158, Closes #153, Closes #151, Closes #150, Closes #149, Closes #146, Closes #133, Closes #132, Closes #31

### CODE REVIEW

File: specs/query/oslc-query.html [hunk 1]
```diff
@@ -409,39 +432,39 @@ <h2>Discovering a Query Capability</h2>
...
-        <p class="conformance">Servers SHOULD support <code>POST</code> on the <i>query base</i> URI of a query
-          capability
-          to avoid clients having to construct URIs that are too long.</p>
+        more query parameters shown in <a href="#Table1">Table 1</a>.
+
       </li>
     </ul>
-
+   
+    <p class="conformance">Servers MUST support a GET on the <i>query base</i> URI of a query capability.</p>
```
Comment (jamsden) [inline specs/query/oslc-query.html]: this should be queryBase

File: specs/query/oslc-query.html [hunk 2]
```diff
@@ -409,39 +432,39 @@ <h2>Discovering a Query Capability</h2>
...
+        
+    <p class="conformance">Servers SHOULD support POST on the query base URI of a query capability to avoid
+      clients having to construct URIs that are too long for a GET.</p>
+      
+    <p class="conformance">A <code>GET</code> or <code>POST</code> request on a query capability MAY use zero, 
+      one or more of the OSLC query parameters listed in <a href="#Table1">Table 1</a>.</p> 
+
+    <p class="conformance">Clients SHOULD use the POST mechanism described above if the URI of a GET
```
Comment (jamsden) [inline specs/query/oslc-query.html]: should not be a conformance clause, we don't specify requirements for clients, only servers

File: specs/query/oslc-query.html [hunk 3]
```diff
@@ -1441,11 +1464,16 @@ <h3>oslc.select</h3>
         behavior.</p>
 
       <p>The syntax of the <code>oslc.select</code> query parameter is the same as that of the
-        <code>oslc.properties</code> query parameter.
-        See <a
-          href="http://docs.oasis-open.org/oslc-core/oslc-core/v3.0/oslc-core-v3.0-part1-overview.html#selectiveProperties">Core
-          3.0 Selective Properties</a> for the syntax.
-
+        <code>oslc.properties</code> query parameter. See [[OSLC-Core-30-Part1]] for the syntax.</p>
```
Comment (jamsden) [inline specs/query/oslc-query.html]: very good, thank you.

**Comment by berezovskyi** [comment]
One more request, after checking the QM working draft: please bump ReSpec to v2.1.15, I guess you are still on .14: https://cdn.jsdelivr.net/gh/oasis-tcs/tab-respec@v2.1.15/builds/respec-oasis-common.min.js

File: specs/query/oslc-query.html (hunk 1 above)
Comment (DavidJHoney) [reply specs/query/oslc-query.html]: Fixed in today's commit.

File: specs/query/oslc-query.html (hunk 2 above)
Comment (DavidJHoney) [reply specs/query/oslc-query.html]: Fixed in today's commit.

File: specs/query/oslc-query.html [hunk 4]
```diff
@@ -129,19 +129,26 @@
...
+        "OSLC-CoreVocab-30": {
           title: "OSLC Core Vocabulary",
           href: "http://docs.oasis-open.org/oslc-core/oslc-core/v3.0/oslc-core-v3.0-part7-core-vocabulary.html",
           authors: ["Jim Amsden", "S. Padgett", "S. Speicher"],
           status: "Working Draft",
           publisher: "OASIS",
         },
+        "OSLC-Discovery-30": {
+          title: "OSLC Core Version 3.0. Part 2: Discovery",
+          href: "http://docs.oasis-open.org/oslc-core/oslc-core/v3.0/oslc-core-v3.0-part2-discovery.html",
```
[the remark this answers is not in the document]
Comment (DavidJHoney) [reply specs/query/oslc-query.html]: Fixed in today's commit.

File: specs/query/oslc-query.html [hunk 5]
```diff
@@ -471,25 +494,18 @@ <h2>Using a Query Capability</h2>
...
-              3.0 Selective Properties</a> for details.</td>
+            <code>oslc.select</code>. See <code>oslc.prefix</code> in [[OSLC-Core-30-Part1]] for details.</td>
         </tr>
         <tr>
           <td><code>oslc.paging</code></td>
           <td>Specifies whether a paged result is requested.
-            See <a
-              href="http://docs.oasis-open.org/oslc-core/oslc-core/v3.0/oslc-core-v3.0-part1-overview.html#resourcePaging">Resource
-              paging</a>
+            See <code>Resource paging</code> in [[OSLC-Core-30-Part1]]
```
[the remark this answers is not in the document]
Comment (DavidJHoney) [reply specs/query/oslc-query.html]: Fixed in today's commit.

File: specs/query/oslc-query.html [hunk 6]
```diff
@@ -586,9 +602,7 @@ <h2>Query Result Containers</h2>
   <section id="QueryResultPaging">
     <h2>Query Result Paging</h2>
 
-    <p class="conformance">A query capability MAY support
-      <a href="http://docs.oasis-open.org/oslc-core/oslc-core/v3.0/oslc-core-v3.0-part1-overview.html#resourcePaging"
-        target="_blank">OSLC Core Resource Paging</a>.</p>
+    <p class="conformance">A query capability MAY support <i>Resource paging</i> as described in [[OSLC-Core-30-Part1]].</p>
```
[the remark this answers is not in the document]
Comment (DavidJHoney) [reply specs/query/oslc-query.html]: Fixed in today's commit.

File: specs/query/oslc-query.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (DavidJHoney) [reply specs/query/oslc-query.html]: Fixed in today's commit.

**Pull request merged**